ports/stm/supervisor/port.c

```diff
@@ -31,25 +31,29 @@
 #include "tick.h"
 
 #include "common-hal/microcontroller/Pin.h"
-#include "common-hal/busio/I2C.h"
-#include "common-hal/busio/SPI.h"
-#include "common-hal/busio/UART.h"
-#include "common-hal/pulseio/PWMOut.h"
-#include "common-hal/pulseio/PulseOut.h"
-#include "common-hal/pulseio/PulseIn.h"
 
-#include "stm32f4/clocks.h"
-#include "stm32f4/gpio.h"
+#if defined(STM32F4)
+    #include "common-hal/busio/I2C.h"
+    #include "common-hal/busio/SPI.h"
+    #include "common-hal/busio/UART.h"
+    #include "common-hal/pulseio/PWMOut.h"
+    #include "common-hal/pulseio/PulseOut.h"
+    #include "common-hal/pulseio/PulseIn.h"
+#endif
 
-#include "stm32f4xx_hal.h"
+#include "clocks.h"
+#include "gpio.h"
 
 safe_mode_t port_init(void) {
     HAL_Init();
     __HAL_RCC_SYSCFG_CLK_ENABLE();
-    __HAL_RCC_PWR_CLK_ENABLE();
+    
+    #if defined(STM32F4)
```

Comment:
Please swap these defines to 1 and 0 so that the conditions aren't dependent on them being defined. Relying on defined means the compiler can't confirm that the correct include chain was actually included.

Reply:
Unfortunately this is not possible, STM32F4 and variant macros STM32F405xx etc are from the ST HAL - the HAL uses #ifdef constantly so I cannot set them to 0 safely. We'd need to make CPY_STM32F4 style versions. 

This might be worth longer discussion. Micropython uses exclusively ifdef(STM32XX) to designate areas as F4/F7/H7 etc. Do you want a new CFLAG we set as 1 or 0 for every variant? That could end up being 10+ CFLAGS that are all 0...

Reply:
@tannewt ping

Reply:
I think that adding new defines are worth the savings from the risk forever more. Just have one place such as mpconfigport.h that defines CIRCUITPY_STM32F4 based on STM32F4 and then use the CIRCUITPY_ version everywhere else. That way it's not CFLAGS.

Reply:
I'm still kind of confused here. So, mpconfigport has a set of duplicate defines, that it sets to 1 or 0 based on the CFLAG ifdef? How does it help the compiler to have a 1 or 0 define in other files if they just started from a CFLAG ifdef in the first place?